presto-parquet/src/main/java/com/facebook/presto/parquet/writer/ParquetWriters.java

```diff
@@ -61,6 +67,25 @@ class ParquetWriters
 {
     private ParquetWriters() {}
 
+    static Optional<ValuesWriterFactory> getValuesWriterFactory(ParquetWriterOptions writerOption)
```

Comment:
It took me quite a while to work through this logic. It;s correct but confusing. I suggest removing all the optional functional gew gaws and instead putting the conditional in the calling method instead of ifPresent